agent/src/main.rs

```diff
@@ -46,26 +45,27 @@ async fn read_hostname() -> Result<String> {
         .to_owned())
 }
 
-async fn read_ssh_pub_host_key(String key_location) -> Result<String> {
-    let public_key = tokio::fs::read_to_string(key_location)
-        .await
+async fn read_ssh_pub_host_key(key_location: String) -> Result<String> {
+    Ok(str::from_utf8(&Sha256::digest(&base64::decode(tokio::fs::read_to_string(key_location).await
         .context("error reading pub key")?
         .trim_end()
         .split(" ")
         .nth(1)
         .to_owned()
-    let pub_key_decoded = base64::decode(public_key).unwrap()
-    let mut sha256_hasher = Sha256::new()
-    sha256_hasher.update(pub_key_decoded)
-    let hashed_result = sha256_hasher.finalize()
-    Ok(str::from_utf8(simple_hex(&hashed_result)).chars().filter(|c| !c.is_whitespace()).collect())
+        .unwrap()
```

Comment:
This chain is actually pretty confusing. Also unwraps aren't cool in production. Can it be split up into locals?